src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py

```diff
@@ -190,6 +190,7 @@ def get_vm_size_completion_list(prefix, action, parsed_args, **kwargs):  # pylin
     register_cli_argument(scope, 'admin_username', help='Username for the VM.', arg_group='Authentication')
     register_cli_argument(scope, 'admin_password', help="Password for the VM if authentication type is 'Password'.", arg_group='Authentication')
     register_cli_argument(scope, 'ssh_key_value', help='SSH public key or public key file path.', completer=FilesCompleter(), type=file_type, arg_group='Authentication')
+    register_cli_argument(scope, 'custom_data', help='Custom init script file or text (cloud-init, cloud-config, etc..)', completer=FilesCompleter(), type=file_type)
```

Comment:
I think the reason behind calling it custom-data is that it acts differently on different OS's. On Linux distros waagent attepts to execute it, but on Windows, it's just a file I believe (https://blogs.msdn.microsoft.com/kristol/2016/05/16/customdata/).